201-decrypt-running-windows-vm-without-aad/azuredeploy.json

```diff
@@ -0,0 +1,50 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vmName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the virtual machine"
+      }
+    },
+    "volumeType": {
+      "type": "string",
+      "defaultValue": "All",
+      "metadata": {
+        "description": "Type of the volume OS or Data to perform encryption operation"
+      }
+    },
+    "forceUpdateTag": {
+      "type": "string",
+      "defaultValue": "1.0",
+      "metadata": {
+        "description": "Pass in an unique value like a GUID everytime the operation needs to be force run"
+      }
+    }
+  },
+  "variables": {

```

Comment:
remove these variables and just use the values in place on the extension

Reply:
Removed 'encryptionOperation' variable. For variables 'extensionName' , 'extensionVersion' I would like to retain them as they can be used to quickly change during deployment to deploy a test extension instead of PROD extension.

Reply:
The name (as we're using it also for the type property) & extension version determines the shape of the resource so not a sound practice to assume one can change without the other.

Reply:
@bmoore-msft  . Didn't get you. This is intentionally left as a variable so that the template can be easily changed during deployment time to deploy extension of type & name 'ADETest' and version '2.3' .

Reply:
That's an uncommon pattern and not one we would recommend users follow.  Very few extensions and RPs work that way.  The type and the version affect the shape of the properties object across ARM.  A change in one of those affects the other.  Changing it is not a simple variable change, it requires intentional changes to the property bag as well.  Types and versions must be explicitly specified.